### PR TITLE
Unify options availabile under CLI and TOML configuration file

### LINKED ISSUE #117: Unify configuration
Currently, the Oracle app uses two forms of configuration: 
1. Command line arguments for:
  - The Owner's private key
  - The Epoch Subrgaph Url
  - Log verbosity
  - a path to the config file
2. A TOML file, for all the rest

Would it be easier to maintain if we transfer tho CLI arguments to the TOML file?

- If so, how should we deal with secrets? Should the whole configuration file be private?
  We made some progress in #116, allowing for environment variables to be used. Should we expand that concept and
  introduce a specific notation for this kind of indirection? _(such as `field = "$VALUE"`)_

- If not, what is the rationale for placing arguments in each place?

### PR DESCRIPTION
**Pull request by neysofu**
This PR fixes #117 and removes all CLI options in favor of settings in the TOML configuration file. The new invocation syntax from CLI is:

`$ block-oracle config.toml`

(Yes, `config.toml` must now be provided as a positional argument, not a flag.)

All examples, as well as Cloudbuild instructions have been updated to reflect these changes. Many values can be provided either using string literals (e.g. `subgraph_url = "https://example.com"`) or environment variables by prefixing the env. var. name with `$` (e.g. `subgraph_url = "$SUBGRAPH_URL"`).

cc @tilacog

### CODE REVIEW

File: crates/oracle/config/staging/config.toml [hunk 1]
```diff
@@ -1,41 +1,44 @@
...
 
+log_level = "trace"
+subgraph_url = "https://api.thegraph.com/subgraphs/name/juanmardefago/block-oracle"
 # FIXME: Once we have access to the Epoch Manager, we can completely remove this entry from configuration.
 # Should be about 10 minutes in Goerli
 epoch_duration = 40
 
 [protocol_chain]
 name = "goerli:0"
 jrpc = "GOERLI_0"
```
Comment (tilacog) [inline crates/oracle/config/staging/config.toml:13]:
```suggestion
jrpc = "$GOERLI_0"
```

**Comment by neysofu** [comment]
I thought I'd updated the CloudBuild instructions, but I accidentally didn't include the changes in the PR. Fixing now.

File: crates/oracle/config/staging/config.toml (hunk 1 above)
Comment (neysofu) [reply crates/oracle/config/staging/config.toml:13]: Good catch, thanks.

**Comment by neysofu** [comment]
Alright, I think I fixed all remaining config files.

**Review by tilacog — APPROVED** [review]
That `FromStrWrapper` is an excellent addition!

**Pull request merged**